### PR TITLE
Merge docs for /java/stream/stream-dynamic.md and /scala/stream/strea…

### PR DESCRIPTION
**Pull request by atiqsayyed**
Ref #23052 
Merge docs for /java/stream/stream-dynamic.md and /scala/stream/stream-dynamic.md

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

File: akka-docs/src/main/paradox/java/stream/stream-dynamic.md [hunk 1]
```diff
@@ -1,128 +1 @@
...
-
-@@snip [HubDocTest.java]($code$/java/jdocs/stream/HubDocTest.java) { #pub-sub-3 }
-
-The resulting Flow now has a type of `Flow[String, String, UniqueKillSwitch]` representing a publish-subscribe
-channel which can be used any number of times to attach new producers or consumers. In addition, it materializes
-to a `UniqueKillSwitch` (see [UniqueKillSwitch](#unique-kill-switch)) that can be used to deregister a single user externally:
-
-@@snip [HubDocTest.java]($code$/java/jdocs/stream/HubDocTest.java) { #pub-sub-4 }
\ No newline at end of file
+../../scala/stream/stream-dynamic.md
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/stream/stream-dynamic.md]: (Like in the other PR's, this still needs some work to make it a 'real' symlink)

**Review by raboof — CHANGES_REQUESTED** [review]
Nice and clean, can be merged after making the Java version a real symlink

**Comment by atiqsayyed** [comment]
Sorry for the silly thing, updated the PR now

**Comment by raboof** [comment]
Nice thanks!

**Pull request merged**